### PR TITLE
chore: restructure pallet part 1, unit tests and internal mockup

### PR DESCRIPTION
**Pull request by neutrinoks**
- moved `tests` into `src`; `mock.rs` and `assets.rs` were combined and also moved to` src`
- in test `balance` applied macros `script_transaction` and `no_type_args` and ExtBuilder`

### CODE REVIEW

**Review by Rqnsom — APPROVED** [review]
LGTM

File: src/tests/balance.rs [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+    ExtBuilder::default()
+        .with_balances(vec![(alice_addr_32.clone(), 10_000)])
+        .build()
+        .execute_with(|| {
+            // Set Alice's balance to a predefined value
+            // assert_ok!(Balances::force_set_balance(
+            //     RuntimeOrigin::root(),
+            //     alice_addr_32.clone(),
+            //     10000,
+            // ));
```
[the remark this answers is not in the document]
Comment (neutrinoks) [reply src/tests/balance.rs]: Done :+1:

**Review by asmie — APPROVED** [review]
Nice! :+1:

File: src/tests/modules.rs [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
-#[test]
 /// Test getting a module.
+#[test]
 fn get_module_correct() {
-    new_test_ext().execute_with(|| {
+    let addr_native = addr32_from_ss58(CAFE_ADDR).unwrap();
+    // let addr_move = AccountAddress::from_hex_literal(CAFE_ADDR).unwrap();
+    // let addr_native = move_to_addr32(&addr_move).unwrap();
+    // println!("{addr_native}");
+    // assert!(false);
```
[the remark this answers is not in the document]
Comment (neutrinoks) [reply src/tests/modules.rs]: Done

File: src/mock.rs [hunk 3]
```diff
@@ -128,87 +142,79 @@ impl ExtBuilder {
...
-    pallet_move_config.assimilate_storage(&mut storage).unwrap();
-
-    storage.into()
-}
-
 // Common constants accross the tests.
-#[allow(dead_code)]
 pub const EMPTY_CHEQUE: u128 = 0; // Not all scripts need the `cheque_amount` parameter.
-pub const CAFE_ADDR: &str = "0xCAFE";
+                                  // pub const CAFE_ADDR: &str = "0xCAFE";
```
[the remark this answers is not in the document]
Comment (neutrinoks) [reply src/mock.rs]: Done

**Review by asmie — APPROVED** [review]

**Pull request merged**